NBitcoin/RPC/RPCClient.cs

```diff
@@ -574,7 +575,12 @@ public async Task<BitcoinAddress> GetRawChangeAddressAsync()
 
 		public Task<RPCResponse> SendCommandAsync(RPCOperations commandName, params object[] parameters)
 		{
-			return SendCommandAsync(commandName.ToString(), parameters);
+			return InternalSendCommandAsync(commandName, parameters, CancellationToken.None);

```

Comment:
What if you make a `CancellationToken ` type check on the last param (and default to None otherwise) then remove it form the params.
Ugly hack I agree

Reply:
Well, I thought about that too before jumping to code and the problem that I saw is that it is a hack (it is not so bad imo) and the IDE cannot provide you the autocomplete help. Anyway, That could have made the changeset a little less aggressive. Let see what @NicolasDorier says, I am not opposed to that change and can do it.

Reply:
:thinking: seems a little obscure for a `public` API, and not very easy to document. Perhaps adding a method overload that includes the `CancellationToken` before the `params` would be good compromise, and also seems to the accepted pattern Microsoft uses in their own codebase (e.g. [1](https://docs.microsoft.com/en-us/dotnet/api/system.data.entity.dbset.findasync?view=entity-framework-6.2.0#System_Data_Entity_DbSet_FindAsync_System_Threading_CancellationToken_System_Object___), [2](https://github.com/dotnet/aspnetcore/blob/v5.0.7/src/JSInterop/Microsoft.JSInterop/src/JSObjectReferenceExtensions.cs#L66))

Reply:
@augustoproiete proposal seems good, it's not a big deal to have an overload and inside call `InternalSendCommandAsync` and no ugly hack.

Reply:
I'm thinking if you go the overload route, you wouldn't need `InternalSendCommandAsync` anymore as the compiler would continue to bind calls to the old method unless a `CancellationToken` is passed through as a second argument keeping the API & ABI backwards compatible.

```diff
public Task<RPCResponse> SendCommandAsync(string commandName, params object[] parameters)
{
-    return InternalSendCommandAsync(commandName, parameters, CancellationToken.None);
+    return SendCommandAsync(commandName, CancellationToken.None, parameters);
 }

-private Task<RPCResponse> InternalSendCommandAsync(string commandName, object[] parameters, CancellationToken cancellationToken)
+public Task<RPCResponse> SendCommandAsync(string commandName, CancellationToken cancellationToken, params object[] parameters)
{
    return SendCommandAsync(new RPCRequest(commandName, parameters), cancellationToken: cancellationToken);
}
```

Reply:
done